src/server.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
+  const parts = normalizedURL.pathname.slice(1).split('/');
+  const query = {};
+
+  for (const [key, value] of normalizedURL.searchParams.entries()) {
+    query[key] = value;
+  }
+
+  if (!parts.length && !query.length) {
+    res.writeHead(400, 'Bad request');
+
+    res.end('No pathname and no searchParams!');
```

Comment:
When you close a request (send a response back to frontend) always add a return statement. 
```suggestion
    return res.end('No pathname and no searchParams!');
```